packages/lesswrong/components/ea-forum/giving-portal/EAGivingPortalPage.tsx

```diff
@@ -321,26 +368,27 @@ const EAGivingPortalPage = ({classes}: {classes: ClassesType<typeof styles>}) =>
   });
    */
   const currentUser = useCurrentUser();
-  const updateCurrentUser = useUpdateCurrentUser();
-  const [notifyForVotingOn, setNotifyForVotingOn] = useState(currentUser?.givingSeasonNotifyForVoting ?? false);
   const {flash} = useMessages();
   const {openDialog} = useDialog();
-  const { captureEvent } = useTracking({ eventProps: { pageContext: "eaGivingPortal" } });
-
-  const toggleNotifyWhenVotingOpens = useCallback(() => {
-    captureEvent('toggleNotifyWhenVotingOpens', { notifyForVotingOn: !notifyForVotingOn })
+  const { electionVote } = useElectionVote("givingSeason23");
+  // We only show the voting banner for users who are eligible -
+  // i.e. those that created their accounts before Oct 23 and haven't voted yet.
+  const showVotingBanner = currentUser && userCanVoteInDonationElection(currentUser) && !electionVote?.submittedAt
 
+  const handleVote = useCallback(() => {
     if (!currentUser) {
       openDialog({
         componentName: "LoginPopup",
         componentProps: {},
-      })
+      });
       return;
     }
-    setNotifyForVotingOn(!notifyForVotingOn);
-    void updateCurrentUser({givingSeasonNotifyForVoting: !notifyForVotingOn});
-    flash(`Notifications ${notifyForVotingOn ? "disabled" : "enabled"}`);
-  }, [captureEvent, notifyForVotingOn, currentUser, updateCurrentUser, flash, openDialog]);
+    if (!userCanVoteInDonationElection(currentUser)) {
+      flash("You are not eligible to vote as your account was created after 22nd Oct 2023");
+      return;
+    }
+    window.location.href = '/voting-portal';
```

Comment:
```suggestion
    navigate('/voting-portal')
```
? Although I'm not actually sure what different behaviour this results in (if any) so I'm going to leave this